### ISSUE TITLE
Connection refused error when calling service endpoint

### ISSUE THREAD

**Issue by WJay-tec** [aws/aws-cloud-map-mcs-controller-for-k8s#273]
**Step to reproduce connection refused error**→

1. Have pods in 2 clusters  → (for example stag-eks , stag-eks-2)
2. Create a ServiceExport for the service you are trying to expose in stag-eks
3. Step 2 will create a ServiceImport on both clusters
4. Create a dummy pod in stag-eks-2, and exec into it. Run a curl command to the ClusterSet endpoint that was exported in step 2 (The curl command will successfully obtain a response)
5. Delete the service pod u created in step 2 in stag-eks
6. Wait for the pod to get recreated, and run the curl command again (which will get a connection refused error)
7. Delete ServiceImport in stag-eks-2 (where the caller is from)
8. Rerun the curl command in the dummy pod in stag-eks-2, and u will get a successful response

Based on my current observation, it seems like coreDNS is not getting the latest pod IP and is still resolving to the old pod ip. When the `ServiceImport` is recreated, it started to work fine again probably because the coreDNS record is updated due to the recreation.

**Comment by runakash** [comment]
@WJay-tec - Thanks for the detailed steps to produce. 

If I understand correctly, your service is ClusterIP?

Can you check the  `ips` in the `ServiceImport` object? If it gets updated after deleting the pod.

```
spec:
  ips:
  - x.x.x.x
  type: "ClusterSetIP"
status:
  clusters:
  - cluster: stag-eks-2
```

**Comment by WJay-tec** [comment]
Yes is ClusterIP, and yes i confirm that the ips in the `ServiceImport` object is updated after deleting the pod.

And i can also confirm that manually deleting the `ServiceImport` after that, my curl command is able to successfully get a response.
